File: MebelOnline.Core/Models/Products/ProductDetailsModel.cs
using MebelOnline.Core.Models.Brands;
using MebelOnline.Core.Models.ProductAttributeValues;
using MebelOnline.Core.Models.ProductImages;
using MebelOnline.Core.Models.ProductOptions;

namespace MebelOnline.Core.Models.Products
{
    public class ProductDetailsModel
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public decimal? OldPrice { get; set; }
        public decimal? Width { get; set; }
        public decimal? Height { get; set; }
        public decimal? Depth { get; set; }
        public BrandModel Brand { get; set; }
        public IList<ProductOptionModel> FrontOptions { get; set; }
        public IList<ProductOptionModel> FrameOptions { get; set; }
        public IList<ProductImageModel> Images { get; set; }
        public IList<ProductAttributeValueModel> Attributes { get; set; }
    }
}

File: mebelonline.client/src/models/productDetailsModel.ts
import type { BrandModel } from './brandModel';
import type { CategoryRefModel } from './categoryDetailsModel';
import type { ProductAttributeValueModel } from './productAttributeValueModel';
import type { ProductImageModel } from './productImageModel';
import type { ProductOptionModel } from './productOptionModel';

export type ProductDetailsModel = {
  id: number;
  title: string;
  description?: string | null;
  price: number;
  oldPrice?: number | null;
  width?: number | null;
  height?: number | null;
  depth?: number | null;
  note?: string | null;
  brand?: BrandModel | null;
  category?: CategoryRefModel | null;
  frontOptions?: ProductOptionModel[] | null;
  frameOptions?: ProductOptionModel[] | null;
  images?: ProductImageModel[] | null;
  attributes?: ProductAttributeValueModel[] | null;
};
